src/base_maths.jl

```diff
@@ -83,39 +83,45 @@ function _check_matmul_dims(A::AbstractMatrix, B::AbstractMatrix)
     ))
 end
 
-function Base.:*(B::BlockDiagonal, M::AbstractMatrix)
+function Base.:*(B::BlockDiagonal{T}, M::Matrix) where T
```

Comment:
Just curious why this was changed to `Matrix` but the next method was not.

Reply:
great catch! I don't think this change shoudl have been included - will fix that up